profiler/src/main/java/com/splunk/opentelemetry/profiler/snapshot/SnapshotVolumePropagator.java

```diff
@@ -25,39 +25,25 @@
 import java.util.Collection;
 import java.util.Collections;
 
-class SnapshotProfilingSignalPropagator implements TextMapPropagator {
-  private static final String PROFILING_SIGNAL = "splunk.trace.snapshot.volume";
-
-  private final TraceRegistry registry;
-
-  SnapshotProfilingSignalPropagator(TraceRegistry registry) {
-    this.registry = registry;
-  }
-
+class SnapshotVolumePropagator implements TextMapPropagator {
   @Override
   public Collection<String> fields() {
-    return Collections.singletonList(PROFILING_SIGNAL);
+    return Collections.emptyList();
   }
 
   @Override
-  public <C> void inject(Context context, C carrier, TextMapSetter<C> setter) {
-    SpanContext spanContext = Span.fromContext(context).getSpanContext();
-    if (registry.isRegistered(spanContext)) {
-      setter.set(carrier, PROFILING_SIGNAL, Volume.HIGHEST.toString());
-    }
-  }
+  public <C> void inject(Context context, C carrier, TextMapSetter<C> setter) {}
 
   @Override
   public <C> Context extract(Context context, C carrier, TextMapGetter<C> getter) {
-    Volume volume = Volume.fromString(getter.get(carrier, PROFILING_SIGNAL));
-    if (volume == Volume.OFF) {
-      return context;
+    if (isTraceRoot(context)) {
+      return context.with(Volume.HIGHEST);
     }
+    return context;
+  }
 
+  private boolean isTraceRoot(Context context) {
     SpanContext spanContext = Span.fromContext(context).getSpanContext();
-    if (spanContext.isValid()) {
-      registry.register(spanContext);
-    }
-    return context;
+    return !spanContext.isValid();
```

Comment:
If you wish to make this shorter you could use `Span.fromContextOrNull(context) == null`